### DISCUSSION TITLE
Doc comment on text2vec-openai (/modules/retriever-vectorizer-modules/text2vec-openai)

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#750] [Docs/blog comments]
# Doc comment on text2vec-openai (/modules/retriever-vectorizer-modules/text2vec-openai)

Overview

https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules/text2vec-openai

**Comment by scepeda78** [comment]
if the content is vectorized with openai, it means that when a query is done, it need to vectorize the query also?

**Reply by kevinamiri** [reply]
Both query and document must be vectorized.

**Comment by devbeekay** [comment]
How can we authenticate with Azure Open AI using active directory method?

**Comment by YanDavKMS** [comment]
I'm trying to use our Azure OpenAI account.

Currently I can only set the following:

"resourceName": "<YOUR-RESOURCE-NAME>",
"deploymentId": "<YOUR-MODEL-NAME>",

But we also need to set the api-version and api-key in order to use our Azure OpenAI account.
How can we do that?

**Reply by ChrisDelClea** [reply]
Totally! I do not get it. How to change the openai-api-base? In my case it always tries to send the requests to: `https://api.openai.com/v1/embeddings`, however, using azure one needs to specify their own URI-Endpoint. Same issue for passing the API-Key , it is not convinient to send it in the headers .

**Comment by cback97** [comment]
How would I get the kmeans clusters of these embeddings from the weaviate collection? I am having a hard time discerning that from the documentation.

**Comment by jiheeyun93** [comment]
I Have a problems using a text2vec-openai ( Azure openai). How can I set a OPENAI_API_BASE, OPENAI_API_VERSION ?
When I used as in the example, the following messages occurs:
dial tcp: lookup azure.openai.azure.com on 127.0.0.11:53: no such host

**Comment by ntermine** [comment]
Hi, I am using the text2vec-openai module in my project with the goal of having weaviate communicate with my LLM locally. Following what is written in the documentation I added the X-OpenAI-BaseURL field both when I create the class and at Query time.
And it works! Or so I thought, very often I end up with the following error:
"get vector input from modules provider: remote client vectorize: unmarshal response body: invalid character '<' looking for beginning of value".

Can anyone help me?

**Comment by dantavo** [comment]
It seems to me there is a bug because the construction of the endpoint to which the Azure OpenAI embedding service is to be invoked requires parameters that are impossible to set.

**Comment by stefanosimonetto** [comment]
Hi! I would like to know if the text2vec-openai module is using the word-to-vec or the transformers to embed a sentence.

**Reply by dudanogueira** [reply]
Hi! text2vec-openai will use this API to vectorize objects:
https://platform.openai.com/docs/guides/embeddings/what-are-embeddings

**Comment by skbom** [comment]
I really would like to use Weaviate but your documentations is outdated. And there is almost no logs that tells me what's wrong when things go wrong. For example, try to not give the OpenAi key docker env. it just runs without throughing any error or warning. I'm not been digging though the code for 2 days to see why it doesn't my AzureOpenAi module not working. Please, if you want dev to use your product atleast test it a bit and keep the docs up to date.
